src/shared/Paramable.ts

```diff
@@ -154,45 +164,60 @@ export class Paramable implements ParamableInterface {
      * checkParameters() instead.
      */
     validate(): ValidationStatus {
-        const status = this.checkParameters()
-        this.isValid = status.isValid
-        if (this.isValid) {
-            this.assignParameters()
+        const realParams: {[key: string]: unknown} = {}
+        for (const prop in this.params) {
+            if (!Object.prototype.hasOwnProperty.call(this.params, prop))
+                continue
+            const p = this.params[prop]
+            if (typeof p.value !== 'string') p.value = `${p.value}`
+
```

Comment:
I figured that for the convenience of visualizer writers the `value` property can still be anything, so that the visualizer writers don't have to bother with converting initial values to strings, especially for ones with specific formatting, like p5 vector. The system will always convert it to a string anyway

Reply:
Where do you stand as far as folding in a type-generic version of the Paramable interface, as modeled by the playground links I posted?

Reply:
We discussed this in weekly meeting; looking forward to the stricter-typing commit and will re-review at that time. Marking this PR as 'draft'; please unmark via "ready for review" when you believe all comments are addressed, thanks!